receiver/signalfxreceiver/signalfxv2_event_to_logdata.go

```diff
@@ -0,0 +1,85 @@
+// Copyright -c Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signalfxreceiver
+
+import (
+	sfxpb "github.com/signalfx/com_signalfx_metrics_protobuf/model"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/splunk"
+)
+
+// SignalFxV2ToMetricsData converts SignalFx proto data points to
+// consumerdata.MetricsData. Returning the converted data and the number of
+// dropped time series.
+func SignalFxV2EventsToLogRecords(
+	logger *zap.Logger,
+	events []*sfxpb.Event,
+) pdata.LogSlice {
+	lrs := pdata.NewLogSlice()
+
```

Comment:
Can we call lrs.Resize() here to reduce the number of allocations? See recommendation here: https://github.com/open-telemetry/opentelemetry-collector/blob/639b9a80ed46ebe1ad244fcf064ed1d9e341056a/consumer/pdata/generated_log.go#L495